.github/workflows/soundness.yml

```diff
@@ -52,10 +52,10 @@ on:
         description: "Container image for the shell check job. Defaults to latest Swift Ubuntu image."
         default: "swift:5.10-noble"
 
-## We are cancelling previously triggered workflow runs
+## Cancel previous runs for the same PR, unless the workflow is running on the main branch, then don't cancel it so we have historical status checks.
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-soundness
-  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref != 'refs/heads/main' || github.run_number }}-soundness
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```

Comment:
What's the point of the group anymore if you're going to make everything (every PR run, and every push on main) have different groups?

If you just want to keep main branch events always running till the end, then you can use something like:

```
${{ github.repository.default_branch == github.ref_name  }}
```
`github.repository.default_branch` is available in most events including pushes and PRs and looks like `main`, and `github.ref_name` will also resolve to 'main' (not available in PR events).

In PRs, you'll need to use `github.base_ref` to get the proper head branch name.
So generally `${{ github.head_ref || github.ref_name }}` will give you the proper branch name in both PRs and push events.

Reply:
> What's the point of the group anymore if you're going to make everything (every PR run, and every push on main) have different groups?

I'd say my ideal setup is one group per PR, and one group per commit on main. I must admit I copied this from an old repo (3-4 years ago?) where we had trouble initially without the run number, but you are right that we probably don't want it here 😅. I'll update accordingly for the default branch.

Reply:
I just realized what I meant to write is 
github.event.repository.default_branch
Not
github.repository.default_branch

The second one will not work 😅

Reply:
Good to know, though no longer necessary since it's been moved only to files that need them 😅